packages/apex-node/src/logs/logService.ts

```diff
@@ -71,27 +78,93 @@ export class LogService {
     });
   }
 
-  public async getLogRecords(numberOfLogs?: number): Promise<LogRecord[]> {
-    let query = 'Select Id, Application, DurationMilliseconds, Location, ';
-    query +=
-      'LogLength, LogUser.Name, Operation, Request, StartTime, Status from ApexLog Order By StartTime DESC';
+  public async getLogById(logId: string): Promise<AnyJson> {
+    const url = `${this.connection.tooling._baseUrl()}/sobjects/ApexLog/${logId}/Body`;
+    const response = (await this.connection.tooling.request(url)) as AnyJson;
+    return { log: response.toString() || '' };
+  }
 
+  public async getLogRecords(numberOfLogs?: number): Promise<LogRecord[]> {
+    let apexLogQuery =
+      'Select Id, Application, DurationMilliseconds, Location, LogLength, LogUser.Name, ' +
+      'Operation, Request, StartTime, Status from ApexLog Order By StartTime DESC';
     if (typeof numberOfLogs === 'number') {
       if (numberOfLogs <= 0) {
         throw new Error(nls.localize('numLogsError'));
       }
       numberOfLogs = Math.min(numberOfLogs, MAX_NUM_LOGS);
-      query += ` LIMIT ${numberOfLogs}`;
+      apexLogQuery += ` LIMIT ${numberOfLogs}`;
     }
 
     const response = (await this.connection.tooling.query(
-      query
+      apexLogQuery
     )) as LogQueryResult;
     return response.records as LogRecord[];
   }
 
+  private async createTraceFlag(userId: string): Promise<void> {
+    const DebugLevelId = await this.getDebugLevelId(DEFAULT_DEBUG_LEVEL_NAME);
+    const startDate = new Date();
+    const expirationDate = new Date(startDate);
+    expirationDate.setMinutes(startDate.getMinutes() + TAIL_LISTEN_TIMEOUT_MIN);
+    const traceFlag = {
+      LogType: LOG_TYPE,
+      TracedEntityId: userId,
+      StartDate: startDate,
+      ExpirationDate: expirationDate,
+      DebugLevelId
+    };
+
+    await this.connection.tooling.create('TraceFlag', traceFlag);
+  }
+
+  private async getDebugLevelId(debugLevel: string): Promise<string> {
+    const DEBUG_LEVEL_QUERY =
+      "SELECT Id FROM DebugLevel WHERE DeveloperName = '%s'";
+    const debugQuery = util.format(DEBUG_LEVEL_QUERY, debugLevel);
+    const debugLevelResult = await this.connection.tooling.query<{
+      Id: string;
+    }>(debugQuery);
+    if (!this.hasRecords(debugLevelResult)) {
+      throw new SfdxError(nls.localize('debugLevelNotFound', debugLevel));
```

Comment:
@AnanyaJha so as I was making the tests for this, I was wondering if this is what we want the command to do. It would be a better user experience to create a DEVELOPER_LOG debug level if we don't have one already, rather than throwing an error in this event (but that would be a different experience than what we have now). What do you think?